src/clj/puppetlabs/services/file_serving/config/puppet_fileserver_config_core.clj

```diff
@@ -0,0 +1,93 @@
+(ns ^{:doc
+      "Tools to parse Puppet fileserver.conf."}
+    puppetlabs.services.file-serving.config.puppet-fileserver-config-core
+  (:require
+    [puppetlabs.kitchensink.core :as ks]
+    [schema.core :as schema]
+    [puppetlabs.puppetserver.ringutils :as ring])
+  (:use [clojure.java.io :only (reader)]
+        [clojure.string :only (split trim split)]))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; internal helpers
+
+(defn- parse-one-line
+  [s]
+  (if (= (first s) \[)
+    (-> s (subs 1 (.indexOf s "]")) trim keyword)
+    (let [n (.indexOf s " ")]
+      (if (neg? n)
+        (throw (Exception. (str "Could not parse: " s)))
+        (let [directive (-> s (subs 0 n) trim keyword)
+              argument (-> s (subs (inc n)) trim)]
+          (if (= :path directive)
+            {:path argument}
+            {:acl [directive argument]}))))))
+
+(defn- strip-comment
+  [s]
+  (let [n (.indexOf s (int \#))]
+    (if (and (not (neg? n)))
+      (subs s 0 n)
+      s)))
+
+(defn- mapify
+  [coll]
+  (loop [xs coll m {} key nil]
+    (if-let [x (first xs)]
+      (if (map? x)
+        (recur (rest xs)
+               (ks/deep-merge-with into m (assoc {} key x))
+               key)
+        (recur (rest xs)
+               (assoc m x {})
+               x))
+      m)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Schemas
+
+(def Mount {:path schema/Str :acl [(schema/pair (schema/either (schema/eq :allow) (schema/eq :deny)) "acl" schema/Str "value")]})
```

Comment:
I don't think this schema actually matches what it's used for, at least based on the test in https://github.com/masterzen/puppet-server/blob/feature/file_metadata/test/unit/puppetlabs/services/file_serving/config/puppet_fileserver_config_core_test.clj#L15-L17

From that  it looks like`find-mount` gets as its first argument

```clojure
 {:files2 {:acl [:allow "127.0.0.1/32" :deny "forbidden-certname.domain.com"]
           :path "/etc/puppet/files2"}
  :files1 {:acl [:allow "127.0.0.1/32" :allow "192.168.10.0/24" :allow "this-certname-only.domain.com"]
           :path "/etc/puppet/files1"}}
```
which should match the `Mounts` schema.

The part that is failing here is the value of `:acl`, for example:

```clojure
[:allow "127.0.0.1/32" :deny "forbidden-certname.domain.com"]
```

I think this schema actually requires that this be in the form 

```clojure
[[:allow "127.0.0.1/32"] [:deny "forbidden-certname.domain.com"]]
```

since `(schema/pair ...)` matches a sequencel, and in the `Mount` schema it is used within a vector.